code/modules/clothing/suits/armor.dm

```diff
@@ -159,6 +156,39 @@
 	src.item_state = "reactiveoff"
 	..()
 
+/obj/item/clothing/suit/armor/reactive/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
+	if(!active)
+		return 0
+	if(prob(hit_reaction_chance))
+		var/mob/living/carbon/human/H = owner
+		visible_message("<span class='danger'>The reactive teleport system flings [H] clear of [attack_text]!</span>")
+		var/list/turfs = new/list()
+		for(var/turf/T in orange(6, H))
+			if(T.density) continue
+			if(T.x>world.maxx-6 || T.x<6)	continue
+			if(T.y>world.maxy-6 || T.y<6)	continue
+			turfs += T
+		if(!turfs.len)
+			turfs += pick(/turf in orange(6, src))
+			var/turf/picked = pick(turfs)
+			if(!isturf(picked)) return
+			if(H.buckled)
+				H.buckled.unbuckle_mob()
+			H.forceMove(picked)
+		return 1
+	return 0
+
+/obj/item/clothing/suit/armor/reactive/fire
+	name = "reactive incendiary armor"
+
+
+/obj/item/clothing/suit/armor/reactive/fire/hit_reaction(owner, attack_text)
+	visible_message("<span class='danger'>The [src] blocks the [attack_text], sending out jets of flame!</span>")
+	for(var/mob/living/carbon/C in orange(6, owner))

```

Comment:
But that could potentially set a mob standing in your square on fire, and thus you because of fire spread!